src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,357 @@
+
+If for example there is a [directive that handles the Fullscreen API](https://github.com/solidjs-community/solid-primitives/blob/main/packages/fullscreen/src/index.ts), it could be tested like this:
+
+```ts title="fullscreen.test.ts"
+import { test, expect, vi } from "vitest"
+import { renderDirective } from "@solidjs/testing-library"
+import { createFullScreen } from "./fullscreen"
+
+test("toggles fullscreen", () => {
+  const targetElement = document.createElement("div")
+  const fs = vi.spyOn(targetElement, "fullscreen")
+  const [setArg, container] = renderDirective(createFullScreen, false)
+  setArg(true)
+  expect(fs).toHaveBeenCalled()
+})
+```
+
+
+## Primitive testing
+
+If the reference to the element is not needed, parts of state and logic can be put into reusable hooks or primitives. Since those do not require any elements, using `render` to test them would require a component that otherwise has no use. To circumvent that, there is a `renderHook` utility that simulates such a component without actually rendering anything.
+
+```ts frame="none"
+const renderResult = renderHook(hook, {
+  initialProps, // an array with arguments being supplied to the hook
+  wrapper, // same as the wrapper optionss for `render`
+})
+const {
+  result, // return value of the hook (mutable, destructuring fixes it)
+  cleanup, // manually remove the traces of the test from the DOM
+  owner, // the owner running the hook to use with `runWithOwner()` 
+} = renderResult
+```
+
+A primitive that manages the state of a counter could be tested like this:
+
+```ts frame="none"
+import { test, expect } from "vitest"
+import { renderHook } from "@solidjs/testing-library"
+import { createCounter } from "./counter"
+
+test("increments count", () => {
+  const { result } = renderHook(createCounter)
+  expect(result.count).toBe(0)
+  result.increment()
+  expect(result.count).toBe(1)
+})
+```
+
+
+## Testing effects
+
+Since effects may happen asynchronous, it can be difficult to test them. `@solidjs/testing-library` comes with a `testEffect` function that takes another function, which will receive a `done` function that can be called once the tests are over. It returns a promise that is resolved once done is called. Errors that would hit the next boundary are used to reject the returned promise.
+
+A test using it might look similar to this:
+
+```ts frame="none"
+const [value, setValue] = createSignal(0)
+return testEffect(done =>
+  createEffect((run: number = 0) => {
+    if (run === 0) {
+      expect(value()).toBe(0)
+      setValue(1)
+    } else if (run === 1) {
+      expect(value()).toBe(1)
+      done()
+    }
+    return run + 1
+  })
+)
+```
+
+## Other testing frameworks
+
+Even though it takes much more configuration, it is still possible to test Solid.js applications in [jest](https://dev.to/lexlohr/testing-your-solidjs-code-2gfh) or [other testing libraries](https://dev.to/lexlohr/testing-solidjs-code-beyond-jest-39p).
+
```

Comment:
Is it possible to include a takeaway from those posts and have it here instead?
We would like to avoid personal content here and have the only links in the docs to be to reference sites or other docs.

Reply:
This would be rather long, since the required configuration would fill about 1-2 screen pages alone, even if you disregard different options for the jest transformation and only include babel, which is the shorter one.

Reply:
I understand that, but I believe we have a recommendation to avoid linking outside of the docs to content that is not mdn or official documentation.  Maybe we could either extend this page with that content or add a separate guide for these testing frameworks. What are your thoughts on this @LadyBluenotes ?

Reply:
Imho, we can probably just remove this section. 

Not necessarily because it's not helpful, but it doesn't really add anything. We can always add support for other testing libraries down the line, if they're requested.

Reply:
I've removed it.